hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieSnapshotExporter.java

```diff
@@ -159,18 +161,85 @@ public void testExportAsHudi() throws IOException {
       assertTrue(dfs.exists(new Path(partition + "/.hoodie_partition_metadata")));
       assertTrue(dfs.exists(new Path(targetPath + "/_SUCCESS")));
     }
+  }
+
+  public static class TestHoodieSnapshotExporterForEarlyAbort extends ExporterTestHarness {
+
+    private HoodieSnapshotExporter.Config cfg;
+
+    @Before
+    public void setUp() throws Exception {
+      super.setUp();
+      cfg = new Config();
+      cfg.sourceBasePath = sourcePath;
+      cfg.targetOutputPath = targetPath;
+      cfg.outputFormat = OutputFormatValidator.HUDI;
+    }
 
     @Test
-    public void testExportEmptyDataset() throws IOException {
+    public void testExportWhenTargetPathExists() throws IOException {
+      // make target output path present
+      dfs.mkdirs(new Path(targetPath));
+
+      // export
+      Throwable t = null;
+      try {
+        new HoodieSnapshotExporter().export(jsc, cfg);
+      } catch (Exception e) {
+        t = e;
+      } finally {
+        assertNotNull(t);
+        assertTrue(t instanceof HoodieSnapshotExporterException);
+        assertEquals("The target output path already exists.", t.getMessage());
```

Comment:
would move to catch ?

Reply:
@leesf The reason of checking the assertion in `finally` is to cover the case when exporter does not throw exception, which should be deemed as failed test. Putting in catch block may still pass in that case. 😄

Reply:
actually it reminds me to use annotation of expected exception `@Test(expected = NullPointerException.class)`. Though i couldn't `assertFalse(dfs.exists(new Path(targetPath)))` in that way, guess it is okay; when exception is thrown, obviously we don't expect target path being created. 
So can I just change these testcases to the annotation way?